src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/PlatformEqtTrace.cs

```diff
@@ -147,6 +175,11 @@ public bool InitializeVerboseTrace(string customLogFile)
         /// <inheritdoc/>
         public bool ShouldTrace(PlatformTraceLevel traceLevel)
         {
+            if (DoNotInitialize)
+            {
+                return false;
+            }
+
             switch (traceLevel)
             {
                 case PlatformTraceLevel.Off:
```

Comment:
Also, once the `TraceSource` is initialized, the values are cached, `Source.Switch.ShouldTrace` is read from cache.